documentation/SolutionsRegistryFormat.md

```diff
@@ -0,0 +1,194 @@
+##Solutions Registry format
+
+### Overall format:
+Each entry in the solution registry should have a unique ID (`Solution.id` in the below example), as well as a name (`name`), and a description of which context it requires to run (`context`). Besides these, information can be provided describing different potential aspects of its lifecycle. This can for example be information about how to start and stop the solution, detect whether it is running, set its settings, etc. These will all be described in the below. The overall structure and allowed keys in a solution description can be seen here.
+
+```
+"Solution.id": {
+    "name": "My Solution"
+    "contexts": { ... },
+    "settingsHandlers": { ... },
+    "configure": [ .. ],
+    "restore": [ .. ],
+    "update": [ .. ],
+    "start": [ .. ],
+    "stop": [ .. ],
+    "isInstalled": [ .. ],
+    
+    // Not yet supported. Post-Cloud4All features.
+    "install": [ ... ],
+    "uninstall": [ ... ],
+    "makeConfigurable": [ ... ],
+    "isRunning": { ... },
```

Comment:
@kaspermarkus

Nit:  "isRunning" is defined as an object here, but on line 154 (R154), it's an array of objects.